helix-core/src/main/java/org/apache/helix/task/JobDispatcher.java

```diff
@@ -94,6 +95,8 @@ public ResourceAssignment processJobStatusUpdateAndAssignment(String jobName,
           workflowResource, jobName, workflowState, jobState));
       finishJobInRuntimeJobDag(_dataProvider.getTaskDataCache(), workflowResource, jobName);
       TaskUtil.cleanupJobIdealStateExtView(_manager.getHelixDataAccessor(), jobName);
+      // New pipeline trigger for workflow status update
```

Comment:
nit, "job status update"?

Reply:
It's about how workflow needs another run to handle "all jobs completed", for example.